dataset/test/dataset_arithmetic_test.cpp

```diff
@@ -686,6 +716,46 @@ TYPED_TEST(DatasetBinaryOpTest, masks_propagate) {
     EXPECT_EQ(res[item].masks()["masks_x"], expectedMask);
 }
 
+TYPED_TEST(DatasetBinaryOpTest, masks_not_shared) {
+  // There are two cases being tested here:
+  // 1. Mask in both operands get ORed, ensure the implementation is not using
+  //    the input variable without copy.
+  // 2. The mask is in one of the operands and gets copied to the output. The
+  //    most sensibly behavior appears to be to *copy* it, i.e., masks behave
```

Comment:
Masks behaving like data (copied). Thoughts:
* Not all masks are constructed/dependent on the data. Consider masks relating to bad detectors between runs for example.
* Typically masks represent a smaller memory overhead than the data (providing sensible management, use of broadcasting etc), so copy behaviour does not seem too problematic. 
* Overall I think the dangers of modifying shared masks is significant enough to think that copying is the right approach. Users will have to get used to constructing/applying common masks if they wish to use identical masks between dataarrays.